pkg/deploy/controller/deployerpod/factory.go

```diff
@@ -42,23 +36,30 @@ func (factory *DeployerPodControllerFactory) Create() controller.RunnableControl
 	deploymentStore := cache.NewStore(cache.MetaNamespaceKeyFunc)
 	cache.NewReflector(deploymentLW, &kapi.ReplicationController{}, deploymentStore, 2*time.Minute).Run()
 
-	// Kubernetes does not currently synchronize Pod status in storage with a Pod's container
-	// states. Because of this, we can't receive events related to container (and thus Pod)
-	// state changes, such as Running -> Terminated. As a workaround, populate the FIFO with
-	// a polling implementation which relies on client calls to list Pods - the Get/List
-	// REST implementations will populate the synchronized container/pod status on-demand.
-	//
-	// TODO: Find a way to get watch events for Pod/container status updates. The polling
-	// strategy is horribly inefficient and should be addressed upstream somehow.
-	podQueue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
-	pollFunc := func() (cache.Enumerator, error) {
-		return pollPods(deploymentStore, factory.KubeClient)
+	// TODO: These should be filtered somehow to include only the primary
+	// deployer pod. For now, the controller is filtering.
+	podLW := &deployutil.ListWatcherImpl{
+		ListFunc: func() (runtime.Object, error) {
+			return factory.KubeClient.Pods(kapi.NamespaceAll).List(deployutil.AnyDeployerPodSelector(), fields.Everything())
```

Comment:
We're not watching pods anywhere else in deployments; did I misunderstand? Which other instances of cross-namespace listing were you referring to? Also note that this is scoped to deployer pods, so at least we're not grabbing all pods across all namespaces.